Functions/QualityControl/xASL_qc_GenerateReport.m

```diff
@@ -0,0 +1,418 @@
+    for iFile=1:size(ExistingPrintFiles)
+        xASL_delete(ExistingPrintFiles{iFile});
+    end
+end
+
+% Set defaults
+settings = xASL_sub_defaultSettings();
+
+% Load Pdf configuration
+config = xASL_adm_LoadPDFConfig(x);
+
+% Print the title
+fprintf('Printing ExploreASL PDF report:   \n');
+
+% Parse the entire Json Stack automatically making all the pages.
+[~] = xASL_sub_parseJson(config, x, [], [], settings);
+
+end
+
+
+%% ================================================================================================================================================
+%% ================================================================================================================================================
+function [settings] = xASL_sub_parseJson(json, x, figure, line, settings)
+    fields = fieldnames(json);
+    
+    for iField=1:length(fields)
+        currentField=json.(fields{iField});
+
+        if strcmp(fields{iField}, 'content') || strcmp(fields{iField}, 'pages')
+            for iContent=1:length(currentField)
+                if iscell(currentField(iContent))
+                    [settings, line] = xASL_sub_parseContent(currentField{iContent}, x, figure, line, settings);
+                else
+                    [settings, line] = xASL_sub_parseContent(currentField(iContent), x, figure, line, settings);
+                end
+            end
+        else
+            [settings, line] = xASL_sub_parseContent(currentField, x, figure, line, settings);
+        end 
+    end
+end
+
+
+function [settings, line] = xASL_sub_parseContent(currentField, x, figure, line, settings)
+
+    if ~isstruct(currentField)
+        return;
+    end
+
+    if ~isfield(currentField, 'type')
+        error([currentField, 'containts no field specifying the type of content']);
+    end
+
+    switch currentField.type
+        case 'text' 
+            line = xASL_sub_PrintText(currentField, figure, line, settings);
+        case 'qc'
+            line = xASL_sub_PrintQC(currentField, x, figure, line, settings);
+        case 'image'
+            settings = xASL_sub_PrintImage(currentField, x, figure, settings);  
+        case 'scan'
+            settings = xASL_sub_PrintScan(currentField, x, figure, settings);
+        case 'page'
+            xASL_sub_printPage(currentField, x, settings);  
+        case 'block'
+            xASL_sub_printBlock(currentField, x, figure, settings);
+        case 'settings'
+            settings = xASL_sub_loadSettings(currentField, settings);  
+        case 'patients' 
+            line = xASL_sub_PrintPatient(x, figure, line, settings);
+    end  
+end
+
+function  xASL_sub_printPage(json, x, settings)
+
+    clear primaryFig ax  
+    primaryFig = figure('visible', 'off','Units', 'centimeters','Position', [0 0 21 29.7]);
+
+    %% Print the Title
+    logoPath = fullfile(x.opts.MyPath, 'Design/ExploreASL_logoHeader.png');
+    [~] = xASL_sub_PrintImage(logoPath, [], primaryFig, settings, [0 0.96 1 0]);
```

Comment:
I am not 100% sure what is meant here.
Do you mean change the variables to <type><Identifier> (figPrimary instead of primaryFig) or do you mean in order or definitions?
For both of these variables only one exists.